File: py/test/test_get_login_info.py
import asyncio
import websockets
import json

async def get_bot_login_info():
    # 你的服务器地址和 Token
    uri = "ws://192.168.31.233:3000?access_token=test"
    
    try:
        # 1. 建立连接
        async with websockets.connect(uri) as websocket:
            print(f"已连接到: {uri}")

            # 2. 接收连接成功的第一个生命周期消息 (lifecycle)
            # 就像你在 log 里看到的那个 meta_event
            init_msg = await websocket.recv()
            print(f"收到初始信号: {init_msg}")

            # 3. 构造请求数据 (Action)
            payload = {
                "action": "get_login_info",
                "params": {},
                "echo": "123"  # 用于追踪请求的 ID
            }

            # 4. 发送请求
            print(f"正在发送 Action: {payload['action']}...")
            await websocket.send(json.dumps(payload))

            # 5. 等待并接收响应
            response = await websocket.recv()
            result = json.loads(response)

            # 6. 解析并美化输出
            print("\n--- 获取成功 ---")
            print(json.dumps(result, indent=4, ensure_ascii=False))
            
            if result.get("status") == "ok":
                user_data = result.get("data", {})
                print(f"\n机器人昵称: {user_data.get('nickname')}")
                print(f"机器人 QQ: {user_data.get('user_id')}")

    except Exception as e:
        print(f"连接或通信发生错误: {e}")

if __name__ == "__main__":
    # 运行异步任务
    asyncio.run(get_bot_login_info())

File: js/test/test_get_login_info.js
const WebSocket = require('ws');

// 配置信息
const url = 'ws://192.168.31.233:3000?access_token=test';

// 创建连接
const ws = new WebSocket(url);

ws.on('open', function open() {
    console.log('✅ 已连接到 NapCat 服务器');

    // 构造请求包
    const requestData = {
        action: 'get_login_info',
        params: {},
        echo: 'node_test_123' // 自定义 echo 方便匹配
    };

    console.log('🚀 正在发送请求...');
    ws.send(JSON.stringify(requestData));
});

ws.on('message', function message(data) {
    // data 是 Buffer 类型，需要转成字符串再解析
    const response = JSON.parse(data.toString());

    // 区分系统事件和 API 响应
    if (response.echo === 'node_test_123') {
        console.log('--- 收到 API 响应 ---');
        console.log(JSON.stringify(response, null, 2));

        if (response.status === 'ok') {
            const { nickname, user_id } = response.data;
            console.log(`\n机器人在线: ${nickname} (${user_id})`);
        }
        
        // 测试完毕后可以手动关闭连接，否则会一直保持监听
        ws.close();
    } else {
        // 这里会收到类似 lifecycle 或者心跳之类的消息
        console.log('📩 收到系统事件:', response.post_type || 'unknown');
    }
});

ws.on('error', function error(err) {
    console.error('❌ 连接错误:', err.message);
});

ws.on('close', () => {
    console.log('🔌 连接已关闭');
});